src/codegen.cpp

```diff
@@ -2495,7 +2504,7 @@ static bool emit_builtin_call(jl_cgval_t *ret, jl_value_t *f, jl_value_t **args,
             if (jl_is_leaf_type(tp0)) {
                 emit_expr(args[1], ctx);
                 assert(jl_is_datatype(tp0));
-                *ret = mark_julia_type(ConstantInt::get(T_size, jl_datatype_nfields(tp0)), false, jl_long_type, ctx);
+                *ret = mark_julia_const(jl_box_long(jl_datatype_nfields(tp0)));
```

Comment:
I think codegen context need a temporary root array for things that might not make it into gen code. Otherwise we could be "bloating" the linfo array with useless things. In practice this probably does not matter that much since most of those values are symbol/small ints/...